documentation/src/test/java/org/hibernate/search/documentation/mapper/orm/bridge/document/model/dsl/dynamic/Book.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.documentation.mapper.orm.bridge.document.model.dsl.dynamic;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.annotations.SortNatural;
+import org.hibernate.search.mapper.pojo.bridge.mapping.annotation.PropertyBinderRef;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.PropertyBinding;
+
+@Entity
+@Indexed
+public class Book {
+
+	@Id
+	private Integer id;
+
+	@ElementCollection
+	@SortNatural
+	@PropertyBinding(binder = @PropertyBinderRef(type = UserMetadataBinder.class))
+	private Map<String, String> userMetadata = new TreeMap<>();
```

Comment:
I was thinking about possible use cases.
I don't know if it is very useful to generate more fields having the exact same mapping and modeling.

Reply:
I'm not sure what you mean by this?

> having the exact same mapping and modeling

Reply:
Ah, you mean using a single template. Well yes it's very limited, but can be useful if all you want to index is text.

A more advanced use case is presented below.